hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -286,14 +288,25 @@ public V get(Object key) {
         return toObject(getInternal(keyData));
     }
 
+    protected Object getInternal(Data keyData) {
+        ClientMessage request = MapGetCodec.encodeRequest(name, keyData, getThreadId());
+        ClientMessage response = invoke(request, keyData);
+        MapGetCodec.ResponseParameters resultParameters = MapGetCodec.decodeResponse(response);
+        return resultParameters.response;
+    }
+
     @Override
-    public <O> O get(Object k, Projection<V, O> p) {
-        // TODO
-        return null;
+    public <O> O get(Object key, Projection<V, O> projection) {
+        checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
+        checkNotNull(projection, NULL_PROJECTION_IS_NOT_ALLOWED);
+
+        Data keyData = toData(key);
+        Data projectionData = toData(projection);
+        return toObject(getInternalWithProjection(keyData, projectionData));
     }
 
-    protected Object getInternal(Data keyData) {
-        ClientMessage request = MapGetCodec.encodeRequest(name, keyData, getThreadId());
+    protected Object getInternalWithProjection(Data keyData, Data projection) {
+        ClientMessage request = MapGetWithProjectionCodec.encodeRequest(name, keyData, projection, getThreadId());
         ClientMessage response = invoke(request, keyData);
         MapGetCodec.ResponseParameters resultParameters = MapGetCodec.decodeResponse(response);
```

Comment:
Need to replace with
`MapGetWithProjectionCodec.ResponseParameters resultParameters = MapGetWithProjectionCodec.decodeResponse(response);`